pkg/imgutil/commit/commit.go

```diff
@@ -356,8 +357,14 @@ func writeContentsForImage(ctx context.Context, snName string, baseImg container
 }
 
 // createDiff creates a layer diff into containerd's content store.
-func createDiff(ctx context.Context, name string, sn snapshots.Snapshotter, cs content.Store, comparer diff.Comparer) (ocispec.Descriptor, digest.Digest, error) {
-	newDesc, err := rootfs.CreateDiff(ctx, name, sn, comparer)
+func createDiff(ctx context.Context, name string, sn snapshots.Snapshotter, cs content.Store, comparer diff.Comparer, compression string) (ocispec.Descriptor, digest.Digest, error) {
+	opts := make([]diff.Opt, 0)
+	mediaType := images.MediaTypeDockerSchema2LayerGzip
+	if compression == "zstd" {
```

Comment:
This should raise an error if the string is invalid

Reply:
add verify this field compression value whether is gzip or zstd.

Reply:
This has to be validated in `pkg/`, not just in `cmd/`